### PR TITLE
Add HOC to access theme context utilities in React class components

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
#### Summary

Add HOC to access theme context utilities in React class components

## Description

Currently we provide the `useTheme` hook to access some theme utilities that can be used to tailor components based on the selected theme.

Since hooks can only be used in functional components, in this PR we're introducing a new HOC which can be used in class components to access the same utilities.

Example:
```jsx
class DummyComponent extends Component {
    render() {
      return (
        <div>
          <p>Current theme is: {this.props.theme}</p>
          <p>Themed value: {this.props.themedValue('foo', 'bar')}</p>
        </div>
      );
    }
  }

  const WrappedComponent = withThemeContext(DummyComponent);
```

### CODE REVIEW

**Review by tdeekens — APPROVED** [review]

**Comment by CarlosCortizasCT** [comment]
@commercetools/shield-team-fe I would appreciate some help with the [build error](https://github.com/commercetools/ui-kit/actions/runs/3968574439/jobs/6801913144).

It complains about a dependency not listed in the `package.json` but I don't see where am I using that dependency.

image

**Comment by tdeekens** [comment]
@CarlosCortizasCT Is it cause you're using JSX now? Could it be that adding react, react-dom and @types/react solves it? It mainly preconsturct's analysis being not happy I think.

File: design-system/package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
   "dependencies": {
     "@babel/runtime": "^7.19.0",
     "@babel/runtime-corejs3": "^7.19.1",
+    "@emotion/react": "^11.4.0",
```
Comment (emmenko) [inline design-system/package.json:28]: Is this needed? 🤔

**Review by emmenko — COMMENTED** [review]
Thanks! 💯

File: design-system/package.json (hunk 1 above)
Comment (emmenko) [reply design-system/package.json:28]: Ah I saw the [comment](https://github.com/commercetools/ui-kit/pull/2391#issuecomment-1400043487) now

**Pull request merged**